homeassistant/components/xiaomi_miio/device.py

```diff
@@ -53,17 +55,20 @@ async def async_connect_device(self, host, token):
         return True
 
 
-class XiaomiMiioEntity(Entity):
+class XiaomiMiioEntity(CoordinatorEntity, Entity):
     """Representation of a base Xiaomi Miio Entity."""
 
-    def __init__(self, name, device, entry, unique_id):
+    def __init__(self, name, device, entry, unique_id, coordinator=None):
         """Initialize the Xiaomi Miio Device."""
+        if coordinator:
```

Comment:
I agree, but I suggest to make this conditional to prevent this PR is becoming to big. Another PR https://github.com/home-assistant/core/pull/51791 that updates the fan platform and also uses this entity class `XiaomiMiioEntity`. For this integration should be merged fist. Further the class `XiaomiMiioEntity` is used in the platforms `air_quality`, `light`, `switch` and `vacuum`.
This PR covers using the coordinator for `humidifier` and the new code in `sensor`. I suggest to leave that for now. I could make an inline remark in the code that the use of the coordinator should be mandantory for all platform setups.

Reply:
Just create a new class for the coordinator entity that also inherits the old base class. We should not make the coordinator optional here.

Reply:
Created an extra specialized class `XiaomiCoordinatedMiioEntity`